### DISCUSSION TITLE
Verbindung zum falschen Access Point

### DISCUSSION THREAD

**Discussion by schlaubstar** [jomjol/AI-on-the-edge-device#2329] [General]
### The Problem

Liebe community,

ich habe folgendes seltsames Problem: 
Mein "Heimnetz" besteht aus 2 FritzBoxen, die ich im Mesh-Betrieb laufen habe.

Beide verwenden also die selbe SSID, das selbe Passwort und spannen ein WLAN im 2,4 bzw. 5GHz Freqzuenzband auf. Soweit ich es sehe, optmieren sie die Frequenzen so, dass sie sich da nicht in die Quere kommen.

In der wlan.ini kann ich ja nur "grobe" Parameter eingeben, vor allem SSID und Passwort.

Bedauerlicherweise verbindet sich mein ESP32 mit AiOnTheEdge aber mit dem schwächsten Access Point und verliert somit die Verbindung immer wieder. Dabei ist ein Raum weiter die "Mesh-Repeater" Fritzbox und zwei weitere ESP Module (allerdings ESP8266, eins im Eigenbau eins in einer Tasmota Steckdose) im gleichen Raum verbinden sich natürlich artig mit dem viel naheliegenderen Accesspoint und haben daher viel bessere Signalstärke.

Ich habe die ESP32CAM bei der "falschen" FritzBox schonmal rausgeschmissen aber sie meldet sich wieder dort an ... mit -82dB 

Habe ich irgendeine Chance hier Einfluss zu nehmen?

### Version

Release: v14.0.3 (Commit: 6c891ab+)

### Logfile

```shell
kein Eintrag vorhanden
```

### Expected Behavior

Eigentlich sollte das ESP32-CAM Modul sich, wie jeder andere Wifi Client, mit dem stärksten AccessPoint (bei dem die SSID + Passwort stimmen) verbinden

**Comment by Kalli01** [comment]
Guten Tag,

bei mir ist das Netzwerk ähnlich aufgebaut. (Fritzbox mit mehren Repeatern und Mesh) und auch hier verbinden sich die ESP nicht immer mit dem Besten Signal.
Ab Version 15.1? gibt es in den Einstellungen unter Expert Mode ganz unten den Punkt  RSSI Threshold. Vielleicht bekommst du das Modul so dazu sich mit einem besserem Signal zu verbinden.

**Comment by schlaubstar** [comment]
Hallo @Kalli01 : Vielen lieben Dank für den Tip. Ich hatte garnicht gesehen, dass wir inzwischen v15 haben.
Habe ein Upgrade gemacht und automatisch hing das ESP Modul nach dem Reboot am richtigen AP. Ich habe den RSSI Threshold mal auf -76db gestellt. Wenn ich die Doku richtig verstehe müsste er dann bei -81db getriggert werden sich einen neuen AP zu suchen. 
Vielen Dank für die Infos

**Reply by jh-pberg** [reply]
Hallo @schlaubstar , hat sich das Modul mit dem nahen AP verbunden? Ich habe  RSSI Threshold geändert, das Modul verbindet sich trotzdem mit dem entfernten AP und nicht mit der FritzBox (1,5 Meter entfernt)
Besten Dank

**Reply by schlaubstar** [reply]
Hi @jh-pberg: Wie begeschrieben hat sich das ESP32 nach dem Update auf die (damals) aktuelle Version sofort mit dem stärksten AccessPoint verbunden, ohne dass ich das Roaming via RSSI Threshold hätte anpassen müssen. Ich habe inzwischen auch schon zweimal den Hauptschalter weg und damit einen "W-LAN Totalausfall" gehabt. Noch immer verbindet sich das ESP32 mit dem stärksten AP. Ich habe aber auch kein "Auge" drauf gehabt.

**Comment by caco3** [comment]
Momentan ist das Wifi Roaming noch by default ausgeschaltet. D.h, dass er den ersten AP verwendet, welchen er findet. Und das ist halt einfach auf gut Glück.

Es ist geplant, das Roaming by default zu aktiveren (bei Neuinstallationen), siehe https://github.com/jomjol/AI-on-the-edge-device/issues/2480

**Comment by wrybit** [comment]
Hatte das Problem auch. Liegt leider am inflexiblen Mesh der AVM Geräte, die ein 11v/11k Standard benötigen.
Hat ein Device keine 11v/11k Protokoll Unterstützung, wählt er die erst beste AP per Zufall. Leider meist falsch.

Der AVM Support hat mir geraten jedem AP eine eigene SSID zu geben und dann dem Device die "richtige" zu nennen.
Mesh funktioniert immer noch und jeder Device hat jetzt den "besten" AP den er nutzen kann. Könnte hier auch helfen.

AVM Doku dazu: https://avm.de/service/wissensdatenbank/dok/FRITZ-Box-7590-AX/3593_WLAN-Gerat-verbindet-sich-nicht-mit-optimalem-Mesh-Repeater?t_id=5736140

Greez Jens

**Comment by schlaubstar** [comment]
Soweit ich weiß bietet die Espressif SDK an, dass man beim Verbindungsaufbau die Mac des Wunsch AP mitgeben kann. Bei dem, ebenfalls auf dem ESP32 lauflaufen Projekt "BSB-Lan' zur Heizungssteuerung, wurde das erst kürzlich implementiert, weil es ähnliche Probleme gab. Wäre evtl. auch hier eine Lösung, wenn man das in der Expert Config abgeben könnte.

**Reply by caco3** [reply]
Funktioniert es denn mit unsener Roamingfunktion nicht?

**Reply by wrybit** [reply]
Ich fürchte, nicht Jeder kann mit diesem Namen "RSSI Threshold" die Funktion dahinter bzw. den Nutzen zuordnen. Vielleicht würde eine mehr Nutzen/Anwender bezogene Bezeichnung die Verständlichkeit erhöhen, gegenüber der sehr präzisen Technik definition?

**Reply by caco3** [reply]
Gerne darfst Du einen Vorschlag machen.
Siehe https://github.com/jomjol/AI-on-the-edge-device-docs/blob/main/param-docs/parameter-pages/System/RSSIThreshold.md

**Comment by wg25** [comment]
Warum dem Client sagen, mit wem er sich verbinden soll? Warum nicht der Infrastruktur mitgeben, an welchen AP sich der Client zu hängen hat? Ahh, FritzBox... ;-)

**Comment by JoeFri2k** [comment]
Hallo also , ich habe aktuell das gleiche Problem, der ESP will sich partou nicht  mit dem AP der 4m weit entfernt ist verbinden. verbindest sich mit dem AP auf dem Dach. 
die threshold funktion habe ich aktivert  jedoch ohne Erfolg.

**Comment by friedpa** [comment]
Hat der 4m entfernte AP auch das 2,4GHz WLAN aufgedreht?

**Reply by JoeFri2k** [reply]
ja bzw. diese ist nur auf 2,4 eingestellt.....
